### ISSUE TITLE
len_trim: error: cannot convert '!llvm.array<1 x i8>' to 'i8'

### ISSUE THREAD

**Issue by sscalpone** [flang-compiler/f18-llvm-project#634]
Needed for test framework.
```
subroutine c (s)
character(*) :: s
i = len_trim(s);
end subroutine
```
```
% bbc ccc.f90
% tco < ccc.mlir |& head
loc("<stdin>":19:9): error: cannot convert '!llvm.array<1 x i8>' to 'i8'
loc("<stdin>":19:9): error: failed to legalize operation 'fir.convert'
```

**Comment by jeanPerier** [comment]
Fixed by #635 that was merged in fir-dev.

**Comment by sscalpone** [comment]
Verified.
